### PR TITLE
Add Version comperator for Github dependencies;

### PR DESCRIPTION
**Pull request by timgluz**
Solution for [#970 - Can't resolve npm GitHub URLs](https://github.com/versioneye/versioneye/issues/670)  and  #88 - add Version comperators for Github

I added a new module group called `Comperators` which in near future could include various implementations of comperators required for `ProjectdependencyService.outdated?`.

The `Comperators::Github` works basically like this:
1. it tries to fetch a commit_date by repo_ref
2. it'll compare the commit_date with the release_date of the latest release.

This PR also updates `PackageParser` with support for Github dependencies as specified by this documentation here - [https://docs.npmjs.com/files/package.json#github-urls](https://docs.npmjs.com/files/package.json#github-urls). Which can be thought demo use case.

For this functionality a parser need to 
 * mark `dep[:version_requested] = 'GITHUB'`
 * extract values for `dep[:repo_fullname]` and `dep[:repo_ref]` fields
 * calling the function needs a valid Github Token.

My idea is that for paying customer you could smt similar:
```
parser = PackageParser.new
auth_token = GithubUpdater.fetch_token_for project
parser.parse_content('json document', auth_token)
```

For not paying customer, you could still pass on `token=nil`, which will make it to ignore checking Github dependencies;

NB! this PR doesn't handle `git://github.com/x/y` version labels for `PackageParser`, neither adds supports to any other Parsers - those should be separate tasks as this PR is already too huge;

### CODE REVIEW

File: lib/versioneye/services/projectdependency_service.rb [hunk 1]
```diff
@@ -206,17 +207,21 @@ def self.release?( projectdependency )
...
+    end
 
     last_update_ago = DateTime.now.to_i - projectdependency.outdated_updated_at.to_i
     return projectdependency.outdated if last_update_ago < A_SECONDS_PER_DAY
 
-    update_outdated!( projectdependency )
+    update_outdated!( projectdependency, product, auth_token )
   end
 
+  #TODO: shas can only include HEX (0-9A-F), mostly 40chars, but maybe as short 7char
```
Comment (reiz) [inline lib/versioneye/services/projectdependency_service.rb]: And what to do then? Take your time and resolve the TODO.

Comment (reiz) [inline lib/versioneye/services/projectdependency_service.rb]: @timgluz I don't like TODOs on a PR. For me that means that the PR is not ready to merge.

**Comment by reiz** [comment]
@timgluz Quiet a big PR. I added some comments. If they are resolved I have to check out that PR on my localhost and run some tests. But I like the idea to have certain parsers only for paying customers :-)

File: lib/versioneye/services/projectdependency_service.rb (hunk 1 above)
Comment (timgluz) [inline lib/versioneye/services/projectdependency_service.rb]: It's for review and it's part of code i didnt implement and i think it's implementation wrong, but i am not sure, there may be reason why it's implemented that way;

**Pull request merged**